### DISCUSSION TITLE
Is there a reason for not using a proprietary documentation generator?

### DISCUSSION THREAD

**Discussion by guilherssousa** [daltonmenezes/interprocess#24] [General]
Title basically explains. Currently, the preject uses its own documentation generator built with Next.js, but there is a reason for not using a generator like Nextra, Docussaurus or Mintlify? Based on the project's development stage, it's way more suitable to use already battle-tested tools to build this parts that are not inherently related to the main project.

I was looking forward to contribute writing the documentations, but during the readings to spot some field to improve, I found that there are a bunch of things, specially Accessibility related like Line heights, ARIA attributes, layout shifts happening on the website.

It woundn't make more sense to use one of these previously mentioned tools to simplify the documenting process, leaving more time to spend on the main library development?

**Comment by daltonmenezes** [comment]
Hi @guilherssousa , I needed something that was in Next.js, at the time Nextra was having a lot of problems and didn't work properly (v1), I looked for other alternatives that would handle Next.js well, but as I wasn't finding it and I didn't want to waste more time looking or hoping for corrections so I decided to make by myself and that fit well with the main page design. Today, maybe it makes sense to change, but it's something I'll need to test and see how it will look in relation to the static and existing site in Next.js, but at the moment I don't have the time to execute this.

**Comment by guilherssousa** [comment]
Can I work on this?

**Comment by daltonmenezes** [comment]
@guilherssousa sure! 💜

**Comment by guilherssousa** [comment]
Just to make sure: Nextra is not too opinionated about styling, and since its internal styling is very is very similar to Tailwind, it makes sense to adopt it too. Any thoughts about it?

**Comment by daltonmenezes** [comment]
@guilherssousa currently the interprocess website and docs uses [stitches](https://stitches.dev/).
You can go with stitches or use tailwind if you prefer to make this migration.

**Comment by guilherssousa** [comment]
Here are some updates on the migration:

I used a new blank Next.js project with Nextra docs and migrating the contents was seamless. These are the things I changed:

- Tailwind instead of Stitches to be able to easily style directly from MDX if necessary
- New texts and design for feature section on home matching Nextra
- New Homepage headline (more engaging, but needs validation)

No worries at this point but there's still work to do, here are the missing points:

- [ ] Fancy code editor on Home with Typewriting effect and Intellisense
- [ ] A fancier background like the previous one (or something like the defaults from [Mintlify](https://react.email/docs/introduction)?)
- [ ] Redo buttons and installing command section to clipboard

You can check progress here (documentation links won't work because of base URLs since project is being hosted on Vercel instead of GH Pages):
- [A scalable and type-safe Electron IPC management tool with enhanced DX - Interprocess](https://interprocess-web.vercel.app/interprocess)
- [Overview - Interprocess](https://interprocess-web.vercel.app/interprocess/docs/getting-started/overview)

**Comment by daltonmenezes** [comment]
Hi @guilherssousa , sorry for the delay. The texts and visual elements must be the same and without changing their appearance. I understand that within the documentation itself some adaptations need to be made, but the visual essence of the main page must be maintained for all pages.

**Comment by daltonmenezes** [comment]
Also, the final version must be able to be hosted as static site on GitHub as the current version.
